src/App.jsx

```diff
@@ -58,7 +66,41 @@ const App = () => {
         "Failed to determine target origin or iframe is not available"
       );
     }
-  }
+  };
+
+  const handleIframeLoad = (iframe) => {
+    const iframeWindow = iframe.contentWindow;
+
+    const messageListener = (event) => {
+      if (event.origin !== new URL(iframe.src).origin) return;
+      if (event.data === "appLoaded" && theme) {
+        window.removeEventListener("message", messageListener);
+        sendMessageToIframe(iframe, { customThemeColors: theme });
+      }
```

Comment:
the theme should not be sent to the child UI's unless it is a custom theme loaded from theme.json. create a separate useState for the custom theme.

Reply:
fixing it!